### PR TITLE
Adding rubygem-kafo_wizards package

### PR DESCRIPTION
**Pull request by mbacovsky**
This package is going to be used in installer and perhaps in hammer. 
Test builds in local mock passed for all scl and nonscl targets.

### CODE REVIEW

File: rubygem-kafo_wizards/rubygem-kafo_wizards.spec [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+URL: https://github.com/theforeman/kafo_wizards
+Source0: http://rubygems.org/downloads/%{gem_name}-%{version}.gem
+
+Requires: %{?scl_prefix_ruby}ruby(rubygems)
+Requires: %{?scl_prefix}rubygem(highline)
+
+%if "%{?scl_ruby}" == "ruby193" || (0%{?el6} && 0%{!?scl:1})
+Requires:      %{?scl_prefix_ruby}ruby(abi)
+BuildRequires: %{?scl_prefix_ruby}ruby(abi)
+BuildRequires: %{?scl_prefix_ruby}rubygems-devel
```
Comment (domcleal) [inline rubygem-kafo_wizards/rubygem-kafo_wizards.spec]: Can be outside of the conditional?

File: rubygem-kafo_wizards/rubygem-kafo_wizards.spec [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+Requires:      %{?scl_prefix_ruby}ruby(release)
+BuildRequires: %{?scl_prefix_ruby}ruby(release)
+BuildRequires: %{?scl_prefix_ruby}rubygems-devel
+%endif
+
+BuildRequires: %{?scl_prefix_ruby}ruby(rubygems)
+BuildArch: noarch
+
+Provides: %{?scl_prefix}rubygem(%{gem_name}) = %{version}
+%{?scl:Obsoletes: ruby193-rubygem-%{gem_name}}
```
Comment (domcleal) [inline rubygem-kafo_wizards/rubygem-kafo_wizards.spec]: Not required, we haven't published this.

File: rubygem-kafo_wizards/rubygem-kafo_wizards.spec [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+%description
+This gem provides tools to define and execute wizard-like user interface for your application. Rendering of the wizards uses highline library.
```
Comment (domcleal) [inline rubygem-kafo_wizards/rubygem-kafo_wizards.spec]: Needs wrapping.

File: rubygem-kafo_wizards/rubygem-kafo_wizards.spec [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+%package doc
+BuildArch:  noarch
+Requires:   %{name} = %{version}-%{release}
+%{?scl:Obsoletes: ruby193-rubygem-%{gem_name}-doc}
```
Comment (domcleal) [inline rubygem-kafo_wizards/rubygem-kafo_wizards.spec]: ditto

**Comment by domcleal** [comment]
Seems to be missing a dependency on highline too.

It might be best if we include this when we come to publishing something that uses it - is that foreman-installer, or kafo itself?

**Comment by domcleal** [comment]
While I remember, this should be added to the blacklists in tito.props for the regular SCL tags, since it's only distributed in non-SCL form.

**Comment by mbacovsky** [comment]
Thanks for review. highline is required on L16, the rest was fixed and PR updated.
I don't mind if we postpone publishing this. Scenarios will introduce dependency on it in Kafo. Networking setup in installer will require it from foreman-installer.

**Comment by mbacovsky** [comment]
I blacklisted the package from regular SCL tags. Originaly I had it listed as I'd like to use wizards from hammer, but it can be enabled anytime later.

**Comment by domcleal** [comment]
> I blacklisted the package from regular SCL tags. Originaly I had it listed as I'd like to use wizards from hammer, but it can be enabled anytime later.

Sure, I'm happy to do it as soon as it's needed, just trying to avoid having unused packages.

Please just comment when there's something that actively depends on this and it can be merged and built.

**Comment by ares** [comment]
headsup: upcoming kafo version will require this, master branch already depends on wizards. There's still some time as Martin want to add diff feature before the 0.7 tagging.

**Comment by ares** [comment]
the diff PR is in final stage before merge, 0.7 will be released soon, @mbacovsky could you please rebase this PR so have have the dependency ready?

File: rubygem-kafo_wizards/rubygem-kafo_wizards.spec [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+Release: 1%{?dist}
+Group: Development/Libraries
+License: GPLv3+
+URL: https://github.com/theforeman/kafo_wizards
+Source0: http://rubygems.org/downloads/%{gem_name}-%{version}.gem
+
+Requires: %{?scl_prefix_ruby}ruby(rubygems)
+Requires: %{?scl_prefix}rubygem(highline)
+
+%if "%{?scl_ruby}" == "ruby193" || (0%{?el6} && 0%{!?scl:1})
```
Comment (domcleal) [inline rubygem-kafo_wizards/rubygem-kafo_wizards.spec]: This should be modernised as ruby193's gone, check the other kafo spec files for an example.

**Comment by mbacovsky** [comment]
PR rebased, scl use updated, test build on koji passed. 

- https://koji.katello.org/koji/taskinfo?taskID=407573 
- https://koji.katello.org/koji/taskinfo?taskID=407571
- https://koji.katello.org/koji/taskinfo?taskID=407570

File: rel-eng/tito.props [hunk 6]
```diff
@@ -22,6 +22,8 @@ blacklist = foreman-installer
   foreman-selinux
   rubygem-kafo
   rubygem-kafo_parsers
+  rubygem-bundler
```
Comment (domcleal) [inline rel-eng/tito.props]: This shouldn't be here.

File: rel-eng/tito.props [hunk 7]
```diff
@@ -37,6 +39,8 @@ blacklist = foreman-installer
...
```
Comment (domcleal) [inline rel-eng/tito.props]: ditto

File: rubygem-kafo_wizards/rubygem-kafo_wizards.spec [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+%if 0%{?scl:1} || (0%{?el6} && 0%{!?scl:1})
+Requires:      %{?scl_prefix_ruby}ruby(abi)
+BuildRequires: %{?scl_prefix_ruby}ruby(abi)
+%else
+Requires:      %{?scl_prefix_ruby}ruby(release)
+BuildRequires: %{?scl_prefix_ruby}ruby(release)
+%endif
+
+Requires: %{?scl_prefix_ruby}ruby(rubygems)
+Requires: %{?scl_prefix_ruby}ruby(highline)
```
Comment (domcleal) [inline rubygem-kafo_wizards/rubygem-kafo_wizards.spec]: `rubygem(highline)`

**Comment by mbacovsky** [comment]
Updated. Issues fixed, scratch builds passed, test install passed.

**Comment by domcleal** [comment]
Thanks @mbacovsky, merged and building.

**Pull request merged**